src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/SpecValidationUtility.cs

```diff
@@ -171,7 +171,7 @@ private static void ValidateFrameworks(PackageSpec spec, IEnumerable<string> fil
                 if (!framework.FrameworkName.IsSpecificFramework)
                 {
                     hasInvalidFrameworks |= true;
-                    var message = string.Format(CultureInfo.CurrentCulture, Strings.SpecValidationInvalidFramework, framework.TargetAlias);
+                    var message = string.Format(CultureInfo.CurrentCulture, Strings.SpecValidationInvalidFramework, framework.TargetAlias, spec.FilePath);
```

Comment:
Set the other properties of the `RestoreLogMessage` as well

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs#L15

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/Errors/RestoreLogMessage.cs#L17

Maybe look for other examples of how that class is instantiated?